java/src/org/openqa/selenium/grid/router/httpd/RouterServer.java

```diff
@@ -141,6 +141,10 @@ protected Handlers createHandlers(Config config) {
         new GraphqlHandler(
             tracer, distributor, queue, serverOptions.getExternalUri(), getServerVersion());
 
+    Boolean disableUi = new RouterOptions(config).disableUi();
+    if (disableUi) {
+      return new Handlers(graphqlRoute("", () -> graphqlHandler), new ProxyWebsocketsIntoGrid(clientFactory, sessions));
```

Comment:
I am not sure this seems right. If we return here, then  don't we end up skipping adding a few things below ? (example: authentication, compliance checks etc.). They don't all seem specific to grid UI but I might be missing some knowledge here. I see something added for Kubernetes too.

Reply:
Thanks @pujagani for the feedback, I made the changes based on this.